src/linear_algebra/span.lean

```diff
@@ -427,6 +427,31 @@ span_eq_bot.trans $ by simp
 
 @[simp] lemma span_zero : span R (0 : set M) = ⊥ := by rw [←singleton_zero, span_singleton_eq_bot]
 
+lemma span_eq_span {R M : Type*} [ring R] [add_comm_group M] [module R M]
```

Comment:
Does this actually need `ring`and `add_comm_group`, or can you relax these typeclasses to semiring and add_comm_monoid?

Reply:
I've tried to remove this actually, but it seems that to cancel `y` from both sides of `v * i * y = 1 * y` we need some notion of subtraction, which in this case comes from `smul_left_injective`, and rewriting it still requires this in one function or another.